File: alica_engine/src/engine/modelmanagement/factories/PreConditionFactory.cpp
#include "engine/modelmanagement/factories/PreConditionFactory.h"
#include "engine/model/PreCondition.h"
#include "engine/modelmanagement/Strings.h"
#include "engine/modelmanagement/factories/ConditionFactory.h"

namespace alica
{
PreCondition* PreConditionFactory::create(const YAML::Node& preConditionNode, AbstractPlan* plan)
{
    auto* preCondition = new PreCondition();
    ConditionFactory::fillCondition(preConditionNode, preCondition, plan);
    preCondition->setEnabled(Factory::getValue<bool>(preConditionNode, alica::Strings::enabled));
    return preCondition;
}
} // namespace alica

File: alica-supplementary/plan-conversion-tool/src/main/java/com/rapyutarobotics/alica/factories/PreConditionFactory.java
package com.rapyutarobotics.alica.factories;

import com.rapyutarobotics.alica.ConversionProcess;
import com.rapyutarobotics.alica.Tags;
import de.unikassel.vs.alica.planDesigner.alicamodel.PreCondition;
import org.w3c.dom.Element;

public class PreConditionFactory extends Factory {

    public static PreCondition create(Element preConditionNode, ConversionProcess cp) {
        PreCondition preCondition = new PreCondition();
        ConditionFactory.fillConditon(preConditionNode, preCondition, cp);
        preCondition.setEnabled(Boolean.parseBoolean(preConditionNode.getAttribute(Tags.ENABLED)));
        return preCondition;
    }
}
